source/documentation/tools/create-a-derived-table/project-structure.md

```diff
@@ -1,30 +1,384 @@
+  - ❌ `stg_[entity].sql` - might be specific enough at first, but will break down in time. Adding the source system into the file name aids in discoverability, and allows understanding where a component model came from even if you aren't looking at the file tree.
+  - ✅ **Plural.** SQL, and particularly SQL in dbt, should read as much like prose as we can achieve. We want to lean into the broad clarity and declarative nature of SQL when possible. As such, unless there’s a single order in your `orders` table, plural is the correct way to describe what is in a table with multiple rows.
+
+### Staging: Models
+
+Now that we’ve got a feel for how the files and folders fit together, let’s look inside one of these files and dig into what makes for a well-structured staging model.
+
+Below, is an example of a standard staging model from one of our models (from `sop_finance_stg__hmpps_general_ledger` model) that illustrates the common patterns within the staging layer. We’ve organized our model into two <Term id='cte'>CTEs</Term>: one pulling in a source table via the [source macro](https://docs.getdbt.com/docs/build/sources#selecting-from-a-source) and the other applying our transformations.
+
+Here we have ordered the fields based on their type, however, you may decide to order you columns differently. See our style guide [link](link to our style guide.) for more details on how you should style your models.
+
+```sql
+-- sop_finance_stg__hmpps_general_ledger.sql
+source as (
+
+    select * from {{ source('sop_finance_stg', 'hmpps_general_ledger') }}
+
+),
+
+renamed as (
+
+    select
+
+        ----------  ids
+        id as cost_centre_id, -- primary key
+        a_id as analysis_code_id,
+        objective as cobjective_code_id,
+
+        ---------- strings
+        extract_version
+
+        ---------- numerics
+        cast(debit_amount / 100.0) as float) as debit_amount,
+        cast(credit_amount / 100.0) as float) as credit_amount,
+        cast(total_amount / 100.0) as float) as total_amount,
+        ---------- booleans
+        is_recoverable,
+
+        ---------- dates
+        date(paid_date) as paid_date,
+
+        ---------- timestamps
+        transction_at
+
+    from source
+
+)
+
+select * from renamed
+```
+
+- Based on the above, the most standard types of staging model transformations are:
+  - ✅ **Renaming**
+  - ✅ **Type casting**
+  - ✅ **Basic computations** (e.g. cents to dollars)
+  - ✅ **Categorizing** (using conditional logic to group values into buckets or booleans, such as in the `case when` statements above)
```

Comment:
See my earlier comment re. surrogate keys. I think in some staging layers this could be a crucial activity

Reply:
added a point abbout generating keys